mutiny-core/src/lib.rs

```diff
@@ -280,39 +302,39 @@ impl<S: MutinyStorage> MutinyWallet<S> {
                 )
                 .await?;
 
-            // now submit the NWC string
-            let nwc_profiles = self.nostr.profiles();
-            let reserved_profile_index = ReservedProfile::MutinySubscription.info().1;
-            let profile_opt = nwc_profiles
-                .iter()
-                .find(|profile| profile.index == reserved_profile_index);
-
-            let nwc_uri = match profile_opt {
-                Some(profile) => {
-                    // profile with the reserved index already exists, do something with it
-                    profile.nwc_uri.clone()
-                }
-                None => {
-                    // profile with the reserved index does not exist, create a new one
-                    let profile = self
-                        .nostr
-                        .create_new_nwc_profile(
-                            ProfileType::Reserved(ReservedProfile::MutinySubscription),
-                            SpendingConditions::RequireApproval,
-                        )
-                        .await?;
-                    profile.nwc_uri
-                }
-            };
-
-            subscription_client.submit_nwc(nwc_uri).await?;
+            // now submit the NWC string if never created before
+            self.ensure_mutiny_nwc_profile(subscription_client).await?;
 
             Ok(())
         } else {
             Err(MutinyError::SubscriptionClientNotConfigured)
         }
     }
 
+    async fn ensure_mutiny_nwc_profile(
+        &self,
+        subscription_client: Arc<subscription::MutinySubscriptionClient>,
+    ) -> Result<(), MutinyError> {
+        let nwc_profiles = self.nostr.profiles();
+        let reserved_profile_index = ReservedProfile::MutinySubscription.info().1;
+        let profile_opt = nwc_profiles
+            .iter()
+            .find(|profile| profile.index == reserved_profile_index);
+        if profile_opt.is_none() {
+            // profile with the reserved index does not exist, create a new one
+            let profile = self
+                .nostr
+                .create_new_nwc_profile(
+                    ProfileType::Reserved(ReservedProfile::MutinySubscription),
+                    SpendingConditions::RequireApproval,
+                )
+                .await?;
+            // only should have to submit the NWC if never created locally before
+            subscription_client.submit_nwc(profile.nwc_uri).await?;
```

Comment:
it is possible we've submitted but lost it locally, will this overwrite?

Reply:
it will overwrite, but it should be the same nwc string because it's kept to a specific index.